src/server/chromium/crBrowser.ts

```diff
@@ -300,12 +311,48 @@ export class CRBrowserContext extends BrowserContext {
     return result;
   }
 
-  async newPageDelegate(): Promise<PageDelegate> {
+  runAfterAllActiveNewPageCalls(callback: () => void) {
+    let count = this._inflightNewPageCount;
+    if (!count) {
+      callback();
+      return;
+    }
+
+    this._newPageResponseListener.add(() => {
+      --count;
+      try {
+        callback();
+      } finally {
+        return !count;
+      }
+    });
+  }
+
+  private _didReceiveNewPageResponse() {
+    const finishedListeners = [];
+    for (const listener of this._newPageResponseListener) {
+      if (listener())
+        finishedListeners.push(listener);
+    }
+    for (const listener of finishedListeners)
+      this._newPageResponseListener.delete(listener);
+  }
+
+  async newPageDelegate(isInternal: boolean): Promise<PageDelegate> {
     assertBrowserContextIsNotOwned(this);
 
     const oldKeys = this._browser.isClank() ? new Set(this._browser._crPages.keys()) : undefined;
 
-    let { targetId } = await this._browser._session.send('Target.createTarget', { url: 'about:blank', browserContextId: this._browserContextId });
+    let targetId;
+    try {
+      ++this._inflightNewPageCount;
+      ({ targetId } = await this._browser._session.send('Target.createTarget', { url: 'about:blank', browserContextId: this._browserContextId }));
```

Comment:
```js
let createTargetPromise;
let targetId;
try {
  createTargetPromise = this._browser._session.send('Target.createTarget', { url: 'about:blank', browserContextId: this._browserContextId }));
  this._pendingCreateTargetPromises.add(createTargetPromise);
  targetId = (await createTargetPromise).targetId;
} finally {
  this._pendingCreateTargetPromises.delete(createTargetPromise);
}
```

Reply:
Unfortunately this doesn't work because `await createTargetPromise` is executed before `awaitActiveNewPageCalls` inside `_onAttachedToTarget` in which case `newPageDelegate` returns before `CRPage` is created. To overcome that we'd need another callback/promise that `CRPage` with given targetId has been created.